plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiBackgroundSplitLoader.java

```diff
@@ -85,8 +87,8 @@ public void run()
             try {
                 future.get();
             }
-            catch (InterruptedException | ExecutionException e) {
-                throw new TrinoException(HUDI_CANNOT_OPEN_SPLIT, "Error generating Hudi split", e);
+            catch (Exception e) {
+                errorListener.accept(e);
```

Comment:
If we don't throw here, someone probably needs to cancel the `splitLoaderFuture` in HudiSplitSource.

Reply:
we are handling and throwing the exception in its caller function @findepi

Reply:
in HudiSplitSource